packages/components/resource-pages/bucket-policy-details-page.tsx

```diff
@@ -0,0 +1,231 @@
+    if (obcLoaded && !obcError && !_.isEmpty(obj)) {
+      const bucketPolicyName = obj?.metadata?.name;
+      const replicationOBs = getReplicationOBCs(obj?.spec?.replicationPolicy);
+      obcData?.forEach((obc) => {
+        obc?.spec?.additionalConfig?.bucketclass === bucketPolicyName &&
+          bpOBCList.push({
+            name: obc?.metadata?.name,
+            ns: obc?.metadata?.namespace,
+          });
+        replicationOBs?.includes(obc?.spec?.bucketName) &&
+          repOBCList.push({
+            name: obc?.metadata?.name,
+            ns: obc?.metadata?.namespace,
+          });
+      });
+    }
+    return [bpOBCList, repOBCList];
+  }, [obj, obcData, obcLoaded, obcError]);
+
+  const policyType: string = obj?.spec?.namespacePolicy?.type;
+  const dataResources = getDataResources(obj);
+  const dataResourceCount = dataResources?.length || 0;
+  const dataResourceLabel = t('{{dataResourceCount}} data source', {
+    dataResourceCount: dataResourceCount,
+  });
+  const obcCount = bucketPolicyOBCList?.length || 0;
+  const obcLabel = t('{{obcCount}} ObjectBucketClaim', {
+    obcCount: obcCount,
+  });
+  const repCount = replicationOBCList?.length || 0;
+  const repLabel = t('{{repCount}} ObjectBucketClaim', {
+    repCount: repCount,
+  });
+
+  return (
+    <>
+      <Modal {...modalProps} />
+      <div className="co-m-pane__body">
+        <SectionHeading text={t('Bucket Details')} />
+        <div className="row">
+          <div className="col-sm-6">
+            <ResourceSummary
+              resource={obj}
+              launchModal={launchModal}
+              resourceModel={NooBaaBucketClassModel}
+            />
+          </div>
+          <div className="col-sm-6">
+            <dt>{t('Status')}</dt>
+            <dd>
+              <BPStatus obj={obj} />
+            </dd>
+            <dt>{t('Policy type')}</dt>
+            <dd>{policyType}</dd>
+            <dt>{t('Data sources')}</dt>
+            <dd>
+              {!!dataResourceCount ? (
+                <DataResourcesPopOver
+                  label={dataResourceLabel}
+                  dataResources={dataResources}
+                />
+              ) : (
+                dataResourceLabel
+              )}
+            </dd>
+            <dt>{t('ObjectBucketClaims')}</dt>
+            <dd>
+              {!!obcCount ? (
+                <OBCPopOver label={obcLabel} obcDetails={bucketPolicyOBCList} />
+              ) : (
+                obcLabel
+              )}
+            </dd>
+            <dt>{t('Replication buckets')}</dt>
+            <dd>
+              {!!repCount ? (
+                <OBCPopOver label={repLabel} obcDetails={replicationOBCList} />
+              ) : (
+                repLabel
+              )}
+            </dd>
+          </div>
+        </div>
+      </div>
+      {!!obcCount && (
+        <GetSecret
+          obj={{
+            metadata: {
+              name: bucketPolicyOBCList[0].name,
```

Comment:
No screenshot for this ? Why the first item only

Reply:
there will be one OBC per bucketClass only for Data federation...
still user can create multiple by mistake, that is why sending [0]th item here so that UI will not break...
we need proper UX to fix this later...